urllib3/response.py

```diff
@@ -351,3 +363,77 @@ def readinto(self, b):
         else:
             b[:len(temp)] = temp
             return len(temp)
+
+    def _read_chunked(self, amt):
+        def readline():
+            r = ""
+            while True:
+                ch = self._fp._safe_read(1)
+                r += ch
+                if r.endswith("\r\n"):
+                    return r
```

Comment:
You are right! (how could I miss that)

On the other hand, this looks pretty ugly:
```python
self._fp.fp.readline()
```
Is there a better way?